main.js

```diff
@@ -199,7 +278,51 @@ entre los proyectos finales que tienen asignados.
 
 Describir el caso:
 
-Mostrar la solucíon en código:
-
+//Proyecto: PetMatcher
+//La función realizada está basada en 2 objetos (perrito y persona) con los cuales a través de propiedades
+ definididas se comparan sus valores, a partir de un número de coincidencias (en este 
+ejemplo: 2), se establecerá si hay match entre la persona y el perrito:
+1. Si el resultado de coincidencias es mayor o igual a 2 se usará un console.log con la leyenda
+"Tu nuevo mejor amigo es X".
+2. En caso contrario, se usará una leyenda con "Por el momento no tenemos el amigo perfecto para tí."
 
+Mostrar la solucíon en código:
 */
+
+let perrito = {
+  name: "Fido",
+  size: "small",
+  color: "black",
+  goodwith: "kids"
+};
+
+let persona = {
+  name: "Nalle",
+  sizePrefered: "small",
+  colorPrefered: "white",
+  goodwithPrefered: "kids"
+};
+
+let perritoValue =Object.values(perrito);
+let personaValue = Object.values(persona);
+let coincidencias = 0;
+
+function match(obj){
+      for(let i = 0; i < perritoValue.length; i++) {
+         if(perritoValue[i] === personaValue[i]) {
```

Comment:
creo que sería bueno que le dieras una leidíta a esto https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

Reply:
const pet = ["Fido", "small", "black", "Kids", "old"];
const preferences = ["Nalle", "Small", "white", "kids", "old"];
var contador = 0;

//Cambiar los elementos del arreglo a mayúsculas
function getUpperCase (pet,preferences) {
  for(var i= 0; i <pet.length; i++){
    pet[i] = pet[i].toUpperCase();
    preferences[i] = preferences[i].toUpperCase();
  }
  console.log(pet, preferences);  
}
getUpperCase(pet, preferences);

//Hacer el match con los elementos en mayúscula
function match (pet, preferences){
     for(let i=0; i < preferences.length; i++) {
        let result = pet.includes(preferences[i]);
            if(result === true) {
                contador += 1;
            }
     }
    if (contador >=3) {
      console.log("Tu nuevo mejor amigo es " + pet[0]);  
      }
      else {
      console.log("Por el momento no tenemos el amigo perfecto para tí")  
      }
}

match(pet, preferences);